### ISSUE TITLE
Federated with Auth0 - Invalid login token. Issuer doesn't match providerName

### ISSUE THREAD

**Issue by thomasyim02** [aws-amplify/amplify-js#9315]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React Native

### Amplify APIs

Authentication

### Amplify Categories

auth

### Environment information

<details>

```
# Put output below this line


```

</details>

### Describe the bug

Unable to federated with Auth0 after following all the Amplify doc.
Error: Invalid login token. Issuer doesn't match providerName.

### Expected behavior

Successful federation, like described in the doc:
https://docs.amplify.aws/lib/auth/advanced/q/platform/js/#federate-with-auth0

### Reproduction steps

1. Follow all the setup from the doc (https://docs.amplify.aws/lib/auth/advanced/q/platform/js/#federate-with-auth0)
- including, OpenID connect setup at IAM and Identity Pool

2. fetch an idToken from Auth0
2. Decode the token for iss (which is the issuer of the token - the domain), and exp (expiration)
3. Federated Login using Amplify Auth
- domain equals iss (https://mydomain.auth0.com/)
- token equals to jwt idToken from Auth0 (checked signature and iss equals to domain using jwt.io)
- expires_at equals to idTokenPayload.exp

### Code Snippet

```javascript

Auth.federatedSignIn(
   idTokenPayload.iss, // https://mydomain.auth0.com/
   {
      token: response.idToken,
      expires_at: idTokenPayload.exp * 1000,
   },
   {
     email: idTokenPayload.email,
   },
)
```

### Log output

<details>

```

Error: Invalid login token. Issuer doesn't match providerName.

```

</details>

### aws-exports.js

```

Amplify.configure({
    Auth: {
      identityPoolId: xxx,
      region: xxx,
      identityPoolRegion: xxx,
      mandatorySignIn: false,
    },
})

```

### Mobile Operating System

Android 11

**Comment by chrisbonifacio** [comment]
Hi @thomasyim02 👋 can you try changing the domain here to not include `https://` like so:

```js
Auth.federatedSignIn(
   "mydomain.auth0.com",
   {
      token: response.idToken,
      expires_at: idTokenPayload.exp * 1000,
   },
   {
     email: idTokenPayload.email,
   },
)
```

**Comment by thomasyim02** [comment]
> Hi @thomasyim02 👋 can you try changing the domain here to not include `https://` like so:
> 
> ```js
> Auth.federatedSignIn(
>    "mydomain.auth0.com",
>    {
>       token: response.idToken,
>       expires_at: idTokenPayload.exp * 1000,
>    },
>    {
>      email: idTokenPayload.email,
>    },
> )
> ```

@chrisbonifacio 
I have tried. It returned another error: domain not registered at identity pool

**Comment by chrisbonifacio** [comment]
<img width="912" alt="Screen Shot 2021-12-06 at 12 07 50 PM" src="https://user-images.githubusercontent.com/16952986/144890196-ac623c23-48fd-4953-b17d-c74e42ae495b.png">

**Comment by thomasyim02** [comment]
> Have you set the domain in your identity pool the same way? I had to remove `https://` from there as well to get it working.
> 
> 

@chrisbonifacio 
Removed https:// or even the slash at the back. No luck,
![image](https://user-images.githubusercontent.com/30122971/144892325-b75431c2-f313-4e65-9ea8-4a29833d895b.png)

**Comment by chrisbonifacio** [comment]
hmm, well it does seem to be recognizing the provider at least... can you share what the request/response looks like from the network activity on the front end when you try to call `Auth.federatedSignIn` with this configuration?

**Comment by thomasyim02** [comment]
@chrisbonifacio 
Really appreciated on reminding me to re-visit the IAM identity provider. I figured out the root cause and need your help to report to the AWS console team. ( yeah i know it's not the amplify team, thanks for the help! ) The issue is with the Provider URL field ( I inputted the back slash that causes the problem, but the cognito idpool console does not show the back slash after the auto-link from IAM)
![image](https://user-images.githubusercontent.com/30122971/144905251-5079e7eb-286d-4902-b057-e146556a8905.png)

The ticket can be closed now. Thanks again and highly appreciated! Spent two days on this issue~
